src/System.Text.Json/tests/Serialization/TestClasses.cs

```diff
@@ -769,6 +769,144 @@ public void Verify()
         }
     }
 
+    public class TestClassWithStringToObjectDictionaries : ITestClass
+    {
+        public Dictionary<string, SimpleTestClass> MyDict { get; set; }
+
+        // TODO: Add support for classes as dictionary values in class-nested dicitonaries for the following types.
+
+        //public IDictionary<string, SimpleTestClass> MyIDict { get; set; }
+        //public IReadOnlyDictionary<string, SimpleTestClass> MyIReadOnlyDict { get; set; }
+        //public ImmutableDictionary<string, SimpleTestClass> MyImmutableDict { get; set; }
+        //public IImmutableDictionary<string, SimpleTestClass> MyIImmutableDict { get; set; }
+        //public ImmutableSortedDictionary<string, SimpleTestClass> MyImmutableSortedDict { get; set; }
+
+        public static readonly byte[] s_data = Encoding.UTF8.GetBytes(
+            @"{" +
+                @"""MyDict"":{" +
+                    @"""Key1"":" + SimpleTestClass.s_json + "," +
+                    @"""Key2"":" + SimpleTestClass.s_json +
+                @"}" +
+            //@"""MyIDict"":{" +
+            //    @"""Key1"":" + SimpleTestClass.s_json + "," +
+            //    @"""Key2"":" + SimpleTestClass.s_json +
+            //@"}" +
+            //@"""MyReadOnlyDict"":{" +
+            //    @"""Key1"":" + SimpleTestClass.s_json + "," +
+            //    @"""Key2"":" + SimpleTestClass.s_json +
+            //@"}" +
+            //@"""MyImmutableDict"":{" +
+            //    @"""Key1"":" + SimpleTestClass.s_json + "," +
+            //    @"""Key2"":" + SimpleTestClass.s_json +
+            //@"}," +
+            //@"""MyIImmutableDict"":{" +
+            //    @"""Key1"":" + SimpleTestClass.s_json + "," +
+            //    @"""Key2"":" + SimpleTestClass.s_json +
+            //@"}," +
+            //@"""MyImmutableSortedDict"":{" +
+            //    @"""Key1"":" + SimpleTestClass.s_json + "," +
+            //    @"""Key2"":" + SimpleTestClass.s_json +
+            //@"}" +
+            @"}");
+
+        public void Initialize()
+        {
+            MyDict = new Dictionary<string, SimpleTestClass>();
+            {
+                SimpleTestClass obj = new SimpleTestClass();
+                obj.Initialize();
+                MyDict["Key1"] = obj;
+            }
+            {
+                SimpleTestClass obj = new SimpleTestClass();
+                obj.Initialize();
+                MyDict["Key2"] = obj;
+            }
+
+            //MyIDict = new Dictionary<string, SimpleTestClass>();
```

Comment:
Are you planning to commit these commented lines?

Reply:
Thanks. Removed.